scripts/c8y-publish-release.sh

```diff
@@ -48,6 +91,20 @@ publish_version() {
     fi
 }
 
+# Set from draft to pre-release
+if [[ $PRE_RELEASE = 1 ]]; then
```

Comment:
Should quote variables and just use single `[]`. Using shellcheck is useful to validate the script.

```suggestion
if [ "$PRE_RELEASE" = 1 ]; then
```

Reply:
Just pushed some minor refactoring by adding a dynamic check if the release has been transitioned from the draft state (e.g. it is better to check for the not being in the draft state as it would also work for pre-releases and normal releases).